File: CS/S171806_CustomArgs.Web/DocumentExportInterceptor.cs
using System.ComponentModel.Composition;
using System.IO;
using DevExpress.Xpf.Printing.Service.Extensions;
using DevExpress.XtraPrinting;

namespace S171806_CustomArgs.Web {
    [Export(typeof(IDocumentExportInterceptor))]
    public class DocumentExportInterceptor : IDocumentExportInterceptor {
        #region IDocumentExportInterceptor Members
        void IDocumentExportInterceptor.InvokeBefore(Document document, ExportOptionsBase exportoptions, object customArgs) {
        }

        void IDocumentExportInterceptor.InvokeAfter(Stream documentContent, ExportOptionsBase exportoptions, object customArgs) {
            var exportState = customArgs as ExportState;
            if(exportState == null) {
                return;
            }
            var fileName = (exportState.FileName ?? "document") + ".pdf";
            using(var file = File.OpenWrite(Path.Combine(StaticInfo.AppDataDirectoryPath, fileName))) {
                documentContent.CopyTo(file);
            }
        }
        #endregion
    }
}

File: VB/S171806_CustomArgs.Web/DocumentExportInterceptor.vb
Imports System.ComponentModel.Composition
Imports System.IO
Imports DevExpress.Xpf.Printing.Service.Extensions
Imports DevExpress.XtraPrinting
Imports Microsoft.VisualBasic

Namespace S171806_CustomArgs.Web
    <Export(GetType(IDocumentExportInterceptor))> _
    Public Class DocumentExportInterceptor
        Implements IDocumentExportInterceptor
#Region "IDocumentExportInterceptor Members"
        Sub InvokeBefore(ByVal document As Document, ByVal exportoptions As ExportOptionsBase, ByVal customArgs As Object) Implements IDocumentExportInterceptor.InvokeBefore
        End Sub

        Sub InvokeAfter(ByVal documentContent As Stream, ByVal exportoptions As ExportOptionsBase, ByVal customArgs As Object) Implements IDocumentExportInterceptor.InvokeAfter
            Dim exportState = TryCast(customArgs, ExportState)
            If exportState Is Nothing Then
                Return
            End If
            Dim fileName = (If(exportState.FileName, "document")) & ".pdf"
            Using file = System.IO.File.OpenWrite(Path.Combine(StaticInfo.AppDataDirectoryPath, fileName))
                documentContent.CopyTo(file)
            End Using
        End Sub
#End Region
    End Class
End Namespace